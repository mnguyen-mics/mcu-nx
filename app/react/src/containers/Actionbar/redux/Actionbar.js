
const ACTION_BAR = {
  BREADCRUMB: {
    PUSH: 'ACTION_BAR_BREADCRUMB_PUSH',
    POP: 'ACTION_BAR_BREADCRUMB_POP',
    SET: 'ACTION_BAR_BREADCRUMB_SET',
  },
  SET_SECONDARY: 'ACTION_BAR_SET_SECONDARY'
};

const setSecondaryActionBar = visible => dispatch =>
    dispatch({
      type: ACTION_BAR.SET_SECONDARY,
      secondaryBar: visible
    });

const pushBreadcrumb = breadcrumbValue => dispatch =>
  dispatch({
    type: ACTION_BAR.BREADCRUMB.PUSH,
    breadcrumb: { breadcrumbName: breadcrumbValue }
  });

const popBreadcrumb = () => dispatch =>
  dispatch({
    type: ACTION_BAR.BREADCRUMB.POP
  });

const setBreadcrumb = (from, path) => dispatch =>
  dispatch({
    type: ACTION_BAR.BREADCRUMB.SET,
    from,
    path
  });


const defaultActionbarState = {
  path: [],
  secondary: null
};


const actionbarState = (state = defaultActionbarState, action) => {
  switch (action.type) {
    case ACTION_BAR.BREADCRUMB.PUSH: {
      const newPath = [...state.path];
      newPath.push(action.breadcrumb);
      return {
        ...state,
        path: newPath
      };
    }

    case ACTION_BAR.BREADCRUMB.POP: {
      const newPath = [...state.path];
      newPath.pop();
      return {
        ...state,
        path: newPath
      };
    }

    case ACTION_BAR.BREADCRUMB.SET: {

      const newPath = action.from === 0 ?
        action.path : state.path.slice(0, action.from).concat(action.path);

      return {
        ...state,
        path: newPath
      };
    }

    case ACTION_BAR.SET_SECONDARY: {
      return {
        ...state,
        secondary: action.secondaryBar
      };
    }

    default:
      return state;
  }
};

export {

  actionbarState,

  setSecondaryActionBar,

  pushBreadcrumb,
  popBreadcrumb,
  setBreadcrumb
};
