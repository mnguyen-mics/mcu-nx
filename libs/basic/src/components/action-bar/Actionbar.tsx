import React from 'react';
import { Row } from 'antd';
import Breadcrumb, { BreadcrumbProps } from 'antd/lib/breadcrumb';
import McsIcon from '../mcs-icon';

const BreadcrumbItem = Breadcrumb.Item;

export interface ActionbarProps extends BreadcrumbProps {
  className?: string;
  edition?: boolean;
  pathItems: React.ReactNode[];
  backgroundColor?: string;
  inverted?: boolean;
}

type Props = ActionbarProps;

class Actionbar extends React.Component<Props> {
  render() {
    const { edition, pathItems, backgroundColor, inverted, className, children, ...rest } =
      this.props;

    return (
      <Row
        align='middle'
        justify='space-between'
        className={`${edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'} ${
          inverted ? 'inverted' : ''
        } ${className ? className : ''}`}
        style={backgroundColor ? { backgroundColor } : {}}
      >
        <Breadcrumb
          className={'mcs-breadcrumb'}
          separator={<McsIcon type='chevron-right' />}
          {...rest}
        >
          {pathItems.map((item, index) => (
            <BreadcrumbItem key={index}>{item}</BreadcrumbItem>
          ))}
        </Breadcrumb>
        <div className='left-part-margin'>{children}</div>
      </Row>
    );
  }
}

export default Actionbar;
