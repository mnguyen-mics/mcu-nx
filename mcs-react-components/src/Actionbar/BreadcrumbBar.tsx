import React from 'react';
import {Link} from 'react-router-dom';
import {Breadcrumb} from 'antd';

import McsIcons from '../McsIcons';

import generateGuid from '../utils/generateGuid';
import {FormattedMessage, injectIntl, InjectedIntlProps} from 'react-intl';
import {BreadcrumbProps} from 'antd/lib/breadcrumb/Breadcrumb';

import * as H from 'history';
import {compose} from 'recompose';

export interface Path {
  name: FormattedMessage.MessageDescriptor,
  path?: H.LocationDescriptor
}

type BreadcrumbBarProvidedProps =
  BreadcrumbBarProps &
  InjectedIntlProps;

export interface BreadcrumbBarProps extends BreadcrumbProps {
  paths: Array<Path>
}

const BreadcrumbBar: React.SFC<BreadcrumbBarProvidedProps> =
   props => {
  const {intl} = props;

  const buildItem = (elt: Path) => {
    const name = intl.formatMessage(elt.name);

    const formatedElt = (name
        ? (name.substr(0, 27) !== name ? `${name.substr(0, 27)}\u2026` : name)
        : null
    );
    const item = elt.path ? <Link to={elt.path}>{formatedElt}</Link> : formatedElt;

    return <Breadcrumb.Item key={generateGuid()}>{item}</Breadcrumb.Item>;
  };

  const sep = <McsIcons type="chevron-right"/>;

  return (
    <Breadcrumb key="breadcrumb" separator={sep} {...props} >
      {props.paths.map(buildItem)}
    </Breadcrumb>
  );
};


export default compose<BreadcrumbBarProvidedProps, BreadcrumbBarProps>(
  injectIntl,
)(BreadcrumbBar);
