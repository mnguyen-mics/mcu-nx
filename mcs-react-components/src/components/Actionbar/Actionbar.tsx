import * as React from 'react';
import { Row } from 'antd';
import * as cuid from 'cuid';
import { compose } from 'recompose';
import { Link } from 'react-router-dom';
import { FormattedMessage, injectIntl, InjectedIntlProps } from 'react-intl';
import Breadcrumb, { BreadcrumbProps } from 'antd/lib/breadcrumb';
import McsIcon from '../Icon';

const BreadcrumbItem = Breadcrumb.Item;

export interface Path {
  name: FormattedMessage.MessageDescriptor | string;
  path?: string;
}

export interface ActionbarProps extends BreadcrumbProps {
  edition?: boolean;
  paths: Path[];
  backgroundColor?: string;
  inverted?: boolean;
}

type Props = ActionbarProps & InjectedIntlProps;

class Actionbar extends React.Component<Props> {
  buildItem = (elt: Path) => {
    const name =
      typeof elt.name === 'string'
        ? elt.name
        : this.props.intl.formatMessage(elt.name);
    const formatedElt = name
      ? name.substr(0, 27) !== name
        ? `${name.substr(0, 27)}\u2026`
        : name
      : null;
    const item = elt.path ? (
      <Link to={elt.path}>{formatedElt}</Link>
    ) : (
      formatedElt
    );

    return <BreadcrumbItem key={cuid()}>{item}</BreadcrumbItem>;
  };

  render() {
    const {
      edition,
      paths,
      intl,
      backgroundColor,
      inverted,
      children,
      ...rest
    } = this.props;
    return (
      <Row
        type="flex"
        align="middle"
        justify="space-between"
        className={`${edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'} ${
          inverted ? 'inverted' : ''
        }`}
        style={backgroundColor ? { backgroundColor } : {}}
      >
        <Breadcrumb
          className={edition ? 'mcs-breadcrumb-edit' : 'mcs-breadcrumb'}
          separator={<McsIcon type="chevron-right" />}
          {...rest}
        >
          {paths.map(this.buildItem)}
        </Breadcrumb>

        <div className="left-part-margin">{children}</div>
      </Row>
    );
  }
}

export default compose<Props, ActionbarProps>(injectIntl)(Actionbar);
