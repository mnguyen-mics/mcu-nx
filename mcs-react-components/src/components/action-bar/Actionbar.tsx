import * as React from 'react';
import { Row } from 'antd';
import Breadcrumb, { BreadcrumbProps } from 'antd/lib/breadcrumb';
import McsIcon from '../mcs-icon';

const BreadcrumbItem = Breadcrumb.Item;

export interface ActionbarProps extends BreadcrumbProps {
  edition?: boolean;
  pathItems: React.ReactNode[];
  backgroundColor?: string;
  inverted?: boolean;
}

type Props = ActionbarProps;

class Actionbar extends React.Component<Props> {

  render() {
    const {
      edition,
      pathItems,
      backgroundColor,
      inverted,
      children,
      ...rest
    } = this.props;

    return (
      <Row
        align="middle"
        justify="space-between"
        className={`${edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'} ${inverted ? 'inverted' : ''
          }`}
        style={backgroundColor ? { backgroundColor } : {}}
      >
        <Breadcrumb
          className={'mcs-breadcrumb'}
          separator={<McsIcon type="chevron-right" />}
          {...rest}
        >
          {pathItems.map((item, index) => <BreadcrumbItem key={index}>{item}</BreadcrumbItem>)}
        </Breadcrumb>
        <div className="left-part-margin">{children}</div>
      </Row>
    );
  }
}

export default Actionbar;
