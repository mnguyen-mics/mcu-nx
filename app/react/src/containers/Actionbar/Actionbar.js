import React from 'react';
import PropTypes from 'prop-types';
import { Row } from 'antd';
import BreadcrumbBar from './BreadcrumbBar';

function Actionbar(props) {
  return (
    <Row
      type="flex"
      align="middle"
      justify="space-between"
      className={'mcs-actionbar-edit'}
    >
      <BreadcrumbBar {...props} className={'mcs-breadcrumb-edit'} />
      <div className="left-part-margin">
        {props.children}
      </div>
    </Row>
  );
}

Actionbar.defaultProps = {
  edition: false,
};

Actionbar.propTypes = {
  edition: PropTypes.bool,
};

export default Actionbar;
