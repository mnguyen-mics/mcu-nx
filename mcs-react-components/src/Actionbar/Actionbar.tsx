import * as React from 'react';
import { Row } from 'antd';
import BreadcrumbBar, {BreadcrumbBarProps} from './BreadcrumbBar';
import {FormattedMessage} from 'react-intl';

export interface ActionbarProps extends BreadcrumbBarProps {
  edition?: boolean,
}

const Actionbar : React.SFC<ActionbarProps> = props  => {

  const { edition } = props;

  return (
    <Row
      type="flex"
      align="middle"
      justify="space-between"
      className={edition ? 'mcs-actionbar-edit' : 'mcs-actionbar'}
    >
      <BreadcrumbBar
        {...props}
        className={edition ? 'mcs-breadcrumb-edit' : 'mcs-breadcrumb'}
      />
      <div className="left-part-margin">
        {props.children}
      </div>
    </Row>
  );
};

export const SimpleActionBar = (title: FormattedMessage.MessageDescriptor) : React.SFC<any> => () => {
  return Actionbar({paths: [{name: title}]})
};


export default Actionbar;
