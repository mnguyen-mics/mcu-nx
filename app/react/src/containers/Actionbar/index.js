import Actionbar from './Actionbar';
import Editbar from './Editbar';
import ActionbarButton from './ActionbarButton';
import SecondaryActionbar from './SecondaryActionbar';

export {
  Actionbar,
  Editbar,
  ActionbarButton,
  SecondaryActionbar
};
