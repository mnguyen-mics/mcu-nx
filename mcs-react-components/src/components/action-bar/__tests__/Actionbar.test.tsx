import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import Actionbar, { ActionbarProps } from '../Actionbar';
import { MemoryRouter } from 'react-router';

it('renders the actionBar', () => {
  const props: ActionbarProps = {
    paths: [
      { name: 'Campaigns', path: 'www.google.fr' },
      { name: 'Display', path: 'www.github.com' },
    ],
  };

  const component = TestRenderer.create(
    <MemoryRouter>
      <Actionbar {...props} />
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
