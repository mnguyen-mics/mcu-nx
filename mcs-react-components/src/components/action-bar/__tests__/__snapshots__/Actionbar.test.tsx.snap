// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the actionBar 1`] = `
<div
  className="ant-row ant-row-space-between ant-row-middle mcs-actionbar "
  style={Object {}}
>
  <div
    className="ant-breadcrumb mcs-breadcrumb"
  >
    <span>
      <span
        className="ant-breadcrumb-link"
      >
        <a
          href="/www.google.fr"
          onClick={[Function]}
        >
          Campaigns
        </a>
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        <span
          className="mcs-icon "
        >
          <i
            className="mcs-chevron-right"
            style={Object {}}
          />
        </span>
      </span>
    </span>
    <span>
      <span
        className="ant-breadcrumb-link"
      >
        <a
          href="/www.github.com"
          onClick={[Function]}
        >
          Display
        </a>
      </span>
      <span
        className="ant-breadcrumb-separator"
      >
        <span
          className="mcs-icon "
        >
          <i
            className="mcs-chevron-right"
            style={Object {}}
          />
        </span>
      </span>
    </span>
  </div>
  <div
    className="left-part-margin"
  />
</div>
`;
